### DISCUSSION TITLE
v1.24.4: ITV disabled

### DISCUSSION THREAD

**Discussion by skovatch** [Ascoware/get-iplayer-automator#429] [Announcements]
I am looking at how to traverse ITVx.com to get all of the available programs. 

I think it's just a matter of time before yt-dlp can be updated to adapt to the itvx.com changes, and all of the content available before the change will still be accessible. Some material will likely be protected or subscription-only and I have no intention of adding decrypting or getting around any paywalls to the application.

**Important note:** this is the last version that will support macOS 10.10. macOS 10.13 will be the new minimum version as of the next update. I don't know right now if the updater can detect your OS and not offer you an update if it's too old; I'll have to check.

**Comment by davidch119** [comment]
I am in Germany using Nord VPN.
Since ITVx went live I can't download anything from ITV not even old programmes.
I can update the cache.
Programme shows as available but it won't download with "problem unknown".
This happened before ITVx but eventually a programme would download.
This of course could be down to ITV blocking my VPN but I have changed the VPN address frequently and I can view programmes on the ITVx website.
I only post this as a comment. It is not of course a complaint. What has happened was what I expected.
I only post it because you implied I should be able to download old programmes.
Hope you find a way to fix things. Good luck.

**Comment by skovatch** [comment]
I have caching of itvx.com shows working again locally. Occasionally the website gets annoyed when you hit it with 1400 http requests from the same IP address so it starts blocking your connections for a while, but that shouldn't be a problem for regular use. From what I'm reading it looks like it will still be some time before anyone on yt-dlp is motivated to update the code to work with ITVx. The 'official' rollout is 8 December even though you can see it now.

I did a quick investigation into using stv.tv for shows but I don't see a good way to get all programs from A-Z or from all categories like you could with itv.com. Scanning one STV page at a time for use with 'Get Current Webpage" to download individual shows is pretty straightforward but you wouldn't be able to auto-record anything. If folks can live with that I'm happy to charge ahead in that direction.

**Reply by cgareth** [reply]
Hi Scott, It goes without saying your work is really appreciated. There was a post you wrote where we could 'buy you a coffee'  - could you point me in that direction again please? b. STV shows are 4 characters long and my version of GiA won't accept them (I'm on 1.22.1); is it my version that's the issue or can I force GiA to accept the prog IDs some other way? I'm more than happy for this to be the way forward, fwiw/fyi.

**Reply by skovatch** [reply]
The current version of GiA has no knowledge of or support for STV.tv -- I have to add that. 
Support is greatly appreciated and you can do so [here.](https://ko-fi.com/scottkovatch) Thanks!
